acceptance_testing.go

```diff
@@ -0,0 +1,836 @@
+	// from the same location as the destination will write to.
+	SourceConfig(*testing.T) map[string]string
+	// DestinationConfig should be a valid config for a destination connector,
+	// writing to the same location as the source will read from.
+	DestinationConfig(*testing.T) map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest(*testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest(*testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions(*testing.T) []goleak.Option
+
+	// WriteToSource receives a slice of records that should be prepared in the
+	// 3rd party system so that the source will read them. The slice will be
+	// used to verify the source connector can successfully execute reads.
+	// It is discouraged for the driver to change the record slice, unless there
+	// is no way to write the records to the 3rd party system, then the slice
+	// should be modified to reflect the expected records a source should read.
+	WriteToSource(*testing.T, *[]Record)
+	// ReadFromDestination should populate the slice with the records that were
+	// written to the destination. The slice will be used to verify the
+	// destination has successfully executed writes.
+	// The capacity of the slice is equal to the number of records that the test
+	// expects to have been written to the destination.
+	ReadFromDestination(*testing.T, *[]Record)
+}
+
+// ConfigurableAcceptanceTestDriver is the default implementation of
+// AcceptanceTestDriver. It provides a convenient way of configuring the driver
+// without the need of implementing a custom driver from scratch.
+type ConfigurableAcceptanceTestDriver struct {
+	Config ConfigurableAcceptanceTestDriverConfig
+}
+
+// ConfigurableAcceptanceTestDriverConfig contains the configuration for
+// ConfigurableAcceptanceTestDriver.
+type ConfigurableAcceptanceTestDriverConfig struct {
+	// Connector is the connector to be tested.
+	Connector Connector
+
+	// SourceConfig should be a valid config for a source connector, reading
+	// from the same location as the destination will write to.
+	SourceConfig map[string]string
+	// DestinationConfig should be a valid config for a destination connector,
+	// writing to the same location as the source will read from.
+	DestinationConfig map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest func(t *testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest func(t *testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+
+	// Skip lets the caller define if any tests should be skipped (useful for
+	// skipping destination/source tests if the connector only implements one
+	// side of the connector)
+	Skip []string
```

Comment:
shouldn't we mention here that the string is a regex?

Reply:
Will update the doc! Actually we need to remove the additional tip about skipping destination/source tests, because that's not needed anymore, the test itself will determine to skip source/destination tests if the connector does not implement it.